src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    let newState = { ...currentState };
+
```

Comment:
you can use object destructing
```suggestion
    const { type, extraData, keysToRemove } = action;
```